src/jit/valuenum.cpp

```diff
@@ -1678,7 +1678,7 @@ ValueNum ValueNumStore::VNForByrefCon(size_t cnsVal)
         Chunk*   c                                             = GetAllocChunk(TYP_BYREF, CEA_Const);
         unsigned offsetWithinChunk                             = c->AllocVN();
         res                                                    = c->m_baseVN + offsetWithinChunk;
-        reinterpret_cast<INT64*>(c->m_defs)[offsetWithinChunk] = cnsVal;
+        reinterpret_cast<size_t*>(c->m_defs)[offsetWithinChunk] = cnsVal;
```

Comment:
I think this should be `target_size_t`?

Reply:
Can the original issue be repro-d on arm?

Reply:
> Can the original issue be repro-d on arm?

I see no reason why it can't be repro on arm, I will try it today.

> I think this should be target_size_t?

When we cross-compile for arm32 on arm64 do we use 4byte pointers? If yes then the signature  needs to be changed to `VNForByrefCon(target_size_tcnsVal)`?

Reply:
Yes, the signature should also be changed to use target_size_t